src/mock_crypto.rs

```diff
@@ -88,19 +90,31 @@ pub mod rust_sodium {
             pub fn sign_detached(m: &[u8], sk: &SecretKey) -> Signature {
                 let mut temp = m.to_vec();
                 temp.extend(&sk.0);
-                Signature(hash256(&temp))
+                Signature(hash128(&temp))
             }
 
             /// Verify the mock signature against the message and the signer's mock public key.
             pub fn verify_detached(signature: &Signature, m: &[u8], pk: &PublicKey) -> bool {
                 let mut temp = m.to_vec();
                 temp.extend(&pk.0);
-                *signature == Signature(hash256(&temp))
+                *signature == Signature(hash128(&temp))
             }
 
-            fn hash256(data: &[u8]) -> [u8; 32] {
-                use tiny_keccak::sha3_256;
-                sha3_256(data)
+            fn hash128(data: &[u8]) -> [u8; 8] {
+                let mut hasher = DefaultHasher::new();
```

Comment:
One of the points was it needs determinism - is this deterministic w.r.t given seeds else reproduction of failure will be impossible ?